src/core/formatters.ts

```diff
@@ -1,7 +1,6 @@
 namespace Plottable {
 
-  export type Formatter = (d: any) => string;
-
+  export type Formatter = (d: any, datum?: any, index?: number, dataset?: Dataset) => string;
```

Comment:
Nice! I'm a little concerned about the lack of type-safety afforded by including the parameters as optional arguments. 


Maybe we can include another `Formatter` type more tightly scoped to the components this change effects, namely `Pie` and `Bar` charts? Consider creating a new formatter type declared under the namespace `Plottable.Plots` called `LabelFormatter` or something which is of type `(d: any, datum: any, index: number, dataset: Dataset) => string;`. Then if you change the type of `_labelFormatter` to `LabelFormatter` in Pie and Bar, we can guarantee that datum, index and dataset will be passed to the labelFormatter callback for those charts while making no promises about the other `Formatter` instances.